src/components/pages/dashboard/profile.js

```diff
@@ -1,10 +1,202 @@
+} from "../../../helpers/user.helper";
+import { useNavigate } from "react-router-dom";
+import { Button, Popconfirm } from "antd";
 import DashboardLayout from "../../layout/dashboard.layout";
 
 export const Profile = () => {
+  const currentUserAccountNumber = getLoggedInUserAccountNumber();
+
+  const registeredUsers = getAllUsers();
+
+  const navigate = useNavigate();
+
+  const [accountName, setAccountName] = useState(getUserAccountName());
+  const [currentPinCode, SetCurrentPinCode] = useState();
+  const [newPinCode, setNewPinCode] = useState();
+  const [confirmNewPinCode, setConfirmNewPinCode] = useState();
+
+  const [checked, setchecked] = useState(false);
+
+  const handleToggleChange = () => {
+    setchecked(!checked);
+  };
+
+  const handleAccountNameChange = (event) => {
+    setAccountName(event.target.value);
+  };
+  const handleCurrentPinChange = (event) => {
+    SetCurrentPinCode(event.target.value);
+  };
+  const handleNewPinChange = (event) => {
+    setNewPinCode(event.target.value);
+  };
+  const handleConfirmNewPineChange = (event) => {
+    setConfirmNewPinCode(event.target.value);
+  };
+
+  const closeUserAccount = () => {
+    const userIndex = getUserIndexByAccountNumber(currentUserAccountNumber);
+    registeredUsers.splice(userIndex, 1);
+    const registeredUsersLeft = JSON.stringify(registeredUsers);
+    localStorage.setItem("MB_USER_ACCOUNTS", registeredUsersLeft);
+    navigate("/");
+  };
+
+  const schema = yup.object({
+    accountName: yup.string().required("Account Name Required"),
+    accountNumber: yup.string(),
+    currentPin: yup.string(),
+    newPin: yup.string().length(4, "Account PIN must be exactly 4 digits"),
+    confirmNewPin: yup
+      .string()
+      .oneOf([yup.ref("newPin"), null], "Pins do not match"),
+  });
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    resolver: yupResolver(schema),
+  });
+  const onSubmit = (data) => {
+    const currentUserIndex = getUserIndexByAccountNumber(
+      currentUserAccountNumber
+    );
+    registeredUsers[currentUserIndex].accountName = data.accountName;
+    if (checked === true && data.currentPin === "") {
+      alert("Insert Pin");
+      return;
+    }
+    if (
+      checked === true &&
+      data.currentPin !== registeredUsers[currentUserIndex].accountPin
+    ) {
+      alert("Pin Incorrect");
+      return;
+    }
+
+    if (checked === true && data.newPin !== data.confirmNewPin) {
+      alert("Pins do not match");
+      return;
+    }
+    if (
+      checked === true &&
+      data.confirmNewPin === registeredUsers[currentUserIndex].accountPin
+    ) {
+      alert("use a different password for update");
+    }
+    if (checked === true && data.newPin === "") {
+      data.newPin = registeredUsers[currentUserIndex].accountPin;
+    }
+    if (checked === true && data.newPin === data.confirmNewPin) {
+      registeredUsers[currentUserIndex].accountPin = data.newPin;
+    }
```

Comment:
Wrap all the checks when `checked` is true so you don't have to repeat checked === true. Also since checked is a boolean variable, you can use `if (checked)` instead of `if (checked === true)`.

Below check is unnecessary. 
```
if (checked === true && data.newPin === "") {
      data.newPin = registeredUsers[currentUserIndex].accountPin;
}
```
If the change PIN checkbox is checked, user must supply new PIN. Otherwise, why would they check the checkbox in the first place? In that case, `newPin` cannot be empty.

In below check, why check against `confirmNewPin` instead of `newPin`? And why is the message so tacky? Isn't "You entered your current PIN. Choose a different PIN" more appropriate? Mind your cases! Also, there's no return statement, so the program will continue to execute when you close the alert box.
```
if (
      checked === true &&
      data.confirmNewPin === registeredUsers[currentUserIndex].accountPin
    ) {
      alert("use a different password for update");
    }
```

Change to this:
```
if (checked) {
   const savedUserPin = registeredUsers[currentUserIndex].accountPin;
   if (!data.currentPin) {
      alert("Current PIN is required");
      return;
   }
   if (data.currentPin !== savedUserPin) {
      alert("Current PIN is incorrect");
      return;
   }
   if (data.newPin !== data.confirmNewPin) {
      alert("PINs do not match");
      return;
   }
   if (data.newPin === savedUserPin) {
      alert("You entered your current PIN. Choose a different PIN");
      return;
   }

   // all is good
   registeredUsers[currentUserIndex].accountPin = data.newPin;
} 

```

Reply:
Done